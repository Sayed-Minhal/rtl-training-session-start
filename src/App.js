function App({name}) {
  return (
    <div className="App">
      <div>Hello, my name is {name}</div>
    </div>
  );
}

export default App;
