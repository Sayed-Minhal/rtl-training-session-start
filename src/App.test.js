import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

test('renders the component and takes snapshot', () => {
  const {container} = render(<App name={'Sayed'}/>);
  expect(container).toMatchSnapshot();
});
test('renders the title', () => {
  render(<App name='Minhal'/>);
  const divWithName = screen.getByText(/Hello, my name is Minhal/i)
  expect(divWithName).toBeInTheDocument();
});

test('should test onSubmit', () => {
  render(<App name='Minhal'/>);
  const formElement = screen.getByTestId('registrationForm');
  fireEvent.submit(formElement, {})
});