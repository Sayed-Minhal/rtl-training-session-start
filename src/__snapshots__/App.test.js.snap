// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component and takes snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div>
      Hello, my name is 
      Sayed
    </div>
    <form
      data-testid="registrationForm"
      name="registrationForm"
    >
      <label
        for="firstName"
      >
        First Name
      </label>
      <input
        id="firstName"
        name="firstName"
        placeholder="Enter your first name"
        type="text"
        value=""
      />
      <label
        for="teamRoleDeveloper"
      >
        Developer
        <input
          id="teamRoleDeveloper"
          name="teamRole"
          placeholder="Select if you are a developer"
          type="radio"
          value="developer"
        />
      </label>
      <br />
      <label
        for="teamRoleTester"
      >
        Tester
        <input
          id="teamRoleTester"
          name="teamRole"
          placeholder="Select if you are a tester"
          type="radio"
          value="tester"
        />
      </label>
      <button
        type="submit"
      >
        Register
      </button>
    </form>
  </div>
</div>
`;
